src/gui/Application.cpp

```diff
@@ -99,56 +99,68 @@ Application::Application(int& argc, char** argv)
     }
     QString identifier = "keepassxc";
     if (!userName.isEmpty()) {
-        identifier.append("-");
-        identifier.append(userName);
+        identifier += "-" + userName;
+    }
 #ifdef QT_DEBUG
-        // In DEBUG mode don't interfere with Release instances 
-        identifier.append("-DEBUG");
+        // In DEBUG mode don't interfere with Release instances
+        identifier += "-DEBUG";
 #endif
-    }
     QString socketName = identifier + ".socket";
     QString lockName = identifier + ".lock";
 
     // According to documentation we should use RuntimeLocation on *nixes, but even Qt doesn't respect
     // this and creates sockets in TempLocation, so let's be consistent.
-    lock = new QLockFile(QStandardPaths::writableLocation(QStandardPaths::TempLocation) + "/" + lockName);
-    lock->setStaleLockTime(0);
-    lock->tryLock();
-    switch (lock->error()) {
+    m_lockFile = new QLockFile(QStandardPaths::writableLocation(QStandardPaths::TempLocation) + "/" + lockName);
+    m_lockFile->setStaleLockTime(0);
+    m_lockFile->tryLock();
+
+    switch (m_lockFile->error()) {
     case QLockFile::NoError:
-        server.setSocketOptions(QLocalServer::UserAccessOption);
-        server.listen(socketName);
-        connect(&server, SIGNAL(newConnection()), this, SIGNAL(anotherInstanceStarted()));
+        // No existing lock was found, start listener
+        m_lockServer.setSocketOptions(QLocalServer::UserAccessOption);
+        m_lockServer.listen(socketName);
+        connect(&m_lockServer, SIGNAL(newConnection()), this, SIGNAL(anotherInstanceStarted()));
         break;
     case QLockFile::LockFailedError: {
-        alreadyRunning = true;
-        // notify the other instance
-        // try several times, in case the other instance is still starting up
         if (config()->get("SingleInstance").toBool()) {
+            // Attempt to connect to the existing instance
             QLocalSocket client;
             for (int i = 0; i < 3; i++) {
                 client.connectToServer(socketName);
                 if (client.waitForConnected(150)) {
+                    // Connection succeeded, this will raise the existing window if minimized
                     client.abort();
+                    m_alreadyRunning = true;
                     break;
                 }
             }
+
+            if (!m_alreadyRunning) {
+                // If we get here then the original instance is likely dead
+                // forceably reset the lock file
+                m_lockFile->removeStaleLockFile();
```

Comment:
@droidmonkey Maybe we could add a `qDebug` or `qWarn` here? Might be useful for debugging purposes.

Reply:
Good idea, I'll add that tonight